projects/epbs-implementation-in-nimbus.md

```diff
@@ -0,0 +1,57 @@
+# `eip-7732: enshrined Proposer-Builder Separation` implementation in the Nimbus Consensus Client
+
+## Motivation
+
+Transitioning from MEV-Boost to enshrined PBS (ePBS) eliminates reliance on third-party software, broadens block construction responsibilities, and ensures block construction adheres to the network's security rules. This integration fosters a transparent and competitive marketplace for block space.
+
+Currently, the system's dependence on a few relays introduces centralization risks and security vulnerabilities, as these relays operate outside Ethereum’s consensus rules. Enshrining PBS within the protocol seeks to mitigate these risks by establishing a secure proposer-builder relationship, encourage direct protocol engagement and create a transparent, equitable MEV market. By reducing reliance on external systems, ePBS enhances decentralization, improves censorship resistance, and supports long-term goals like MEV redistribution mechanisms (e.g., MEV-burn).
+
+## Project description
+
+This project aims to integrate proposer-builder interaction, where block proposers delegate their block building rights to specialized block builders to maximize extractable value by optimally ordering transactions within the protocol. By separating block building from block proposal, the Maximal Extractable Value (MEV) is distributed across more validators, rather than being concentrated with the most effective MEV searchers. Allowing specialized block builders to operate also reduces the resource burden on individual participants, enabling more independent validators to verify the honesty of blocks. This project seeks to implement proposer-builder separation in the Nimbus Consensus client, following the specifications of the [EIP-7732](https://eips.ethereum.org/EIPS/eip-7732#abstract) document.
+
+## Specification
+
+This specification involves decoupling the execution from consensus and the notion of splitting the slot into 2 parts
+All major changes will be happening in the Consesnus client as specified in this [eip](https://eips.ethereum.org/EIPS/eip-7732). 
+
+## Roadmap
+This below schedule might not hold strictly and the proposed two-weeks-per-component schedule might prove difficult, and I suspect it might prove necessary to interleave things more and difficult to try to plan too tightly. However, the main challenges are listed in this roadmap:
+
+- _Week 5-6_: Understand the current Nimbus-eth2 architecture with the [`storeBlock`](https://github.com/status-im/nimbus-eth2/blob/unstable/beacon_chain/gossip_processing/block_processor.nim `storeBlock`) procedure as a good entry point while getting familiar with the Nim language<br>
+- _Week 7-8_: Implement beacon change specs and write tests <br>
+- _Week 9-10_: Implement the fork-choice specs and write tests <br>
+- _Week 11-12_: Implement the p2p specification changes for ePBS and write tests <br>
+- _Week 13-14_: Implement the honest validator and builder specs with accompanying tests <br>
```

Comment:
Hey @Tomi-3-0, The spec until here is already written, just wanted to highlight it so you can focus more on implementing PoC and not re-write specs

Reply:
Do you mean already implemented in Prysm?

Reply:
Hey @Tomi-3-0, No the spec here consensus spec is same for all the client implementation, unless there is any change you want to implement in the spec mentioned in EIP-7732, it is safe to follow the spec

Reply:
> Hey @Tomi-3-0, No the spec (here consensus spec) is same for all the client implementation, unless there is any change you want to implement in the spec mentioned in EIP-7732, it is safe to follow the spec

So, basically this is just a roadmap based on the outlined spec in the eip.
More like a guide for me on what I should be focusing on at a particular week. I'm not reinventing anything

Reply:
okay makes sense, just wanted to point out to see if it is intentional or not, All the best!